src/libraries/Microsoft.Extensions.DependencyInjection.Specification.Tests/src/KeyedDependencyInjectionSpecificationTests.cs

```diff
@@ -152,10 +153,48 @@ public void ResolveKeyedServicesAnyKeyWithAnyKeyRegistration()
             _ = provider.GetKeyedService<IService>("something-else");
             _ = provider.GetKeyedService<IService>("something-else-again");
 
-            // Return all services registered with a non null key, but not the one "created" with KeyedService.AnyKey
+            // Return all services registered with a non null key, but not the one "created" with KeyedService.AnyKey,
+            // nor the KeyedService.AnyKey registration
             var allServices = provider.GetKeyedServices<IService>(KeyedService.AnyKey).ToList();
-            Assert.Equal(5, allServices.Count);
-            Assert.Equal(new[] { service1, service2, service3, service4 }, allServices.Skip(1));
+            Assert.Equal(4, allServices.Count);
+            Assert.Equal(new[] { service1, service2, service3, service4 }, allServices);
```

Comment:
```suggestion
            Assert.Equal(new[] { service1, service2, service3, service4 }, allServices);
            
            var someKeyedServices = provider.GetKeyedServices<IService>("service").ToList();
            Assert.Equal(new[] { service2, service3, service4 }, someKeyedServices);
            
            var unkeyedServices = provider.GetServices<IService>().ToList();
            Assert.Equal(new[] { service5, service6 }, unkeyedServices);
```

Nit: I'm assuming this all works, and some of this might even be covered partially by other tests. But we don't have many tests with the combination of `AnyKey` registrations, unkeyed registrations, and enumerable resolution of both keyed and unkeyed services. And I don't see the harm in extra verification.